mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -1191,6 +1279,94 @@ mod tests {
         }
     }
 
+    #[test]
+    fn it_batches_ok_when_leftover_from_file_and_write_only_wrap_around() {
```

Comment:
Just to confirm I am understanding this, this is doing the same thing to the ring buffer as the other test you added. But it is testing the batching?

Reply:
It is a very similar test, but instead of testing that the pointers are where they should be I am verifying can still read. Will update this further with different data inserted to verify data is correct.